### ISSUE TITLE
[SR-15431] Support DocC references to symbols defined in another module

### ISSUE THREAD

**Issue by Mordil** [apple/swift-docc#208]

**Comment by maweefeng** [comment]
any news for this ?

**Comment by flockoffiles** [comment]
I would also be very much interested in having this support implemented.

**Comment by spnkr** [comment]
👍

**Comment by Vitalii-Gozhenko** [comment]
+1 for this feature. Sometimes it is incredible usable

**Comment by project-academy** [comment]
+1 for this feature!

**Comment by gposcidonio** [comment]
Big +1 gang.

**Comment by SomeRandomiOSDev** [comment]
+2 if allowed, otherwise +1. Been looking for a way to do this for quite some time

**Comment by extremereality2** [comment]
+1 for this feature

**Comment by ricocrescenzio95** [comment]
+1 for this guyssss

**Comment by RemiBardon** [comment]
Any news regarding implementation?

**Comment by KirillLezhninRMR** [comment]
+1 for this feature

**Comment by d-ronnqvist** [comment]
This feature is very important to us and to DocC but because of how central it is to the broader support for combined documentation for multiple modules, it needs to be thoroughly thought through so that there is a good path forward for future directions and so that we're not designing ourselves into a corner further down the line.



I find the thread of an individual issue to be both too broad and too narrow for the various discussions around a feature like this. The [Swift Forums][forum] is a better place for that. That's where we've [gathered inputs for use cases][input] and [discussed a proposed high level design][plan] for this. That design remains largely unchanged although some initial work have revealed a few areas that require further consideration to ensure that we have enough flexibility going forward without needing to design absolutely everything up front.

[forum]: https://forums.swift.org
[input]: https://forums.swift.org/t/use-cases-for-combined-documentation-of-multiple-targets-in-swift-docc/59319
[plan]: https://forums.swift.org/t/combined-documentation-of-multiple-targets/59579


I should follow up in the Swift Forums with additional details about the plan.


I'll also think about ways, for example keywords, to indicate what issues and pull requests relate to multi-module documentation for those developers who want to follow along in the repo or who want to contribute.

**Comment by d-ronnqvist** [comment]
I want to emphasize that it's _very_ early but we are making some progress on this front with https://github.com/apple/swift-docc/pull/710. 

Even when that PR is merged, the overall user experience will be incomplete until we also support combining documentation from multiple targets into a unified archive.

**Comment by joshuapoq** [comment]
Very excited to come across this! Have been finding it frustrating that most guides or posts specify adding DocC to a 'framework or package' or using the 'package name', mixing terminology making it harder to figure out how to add DocC to a Swift Package of many products.

Based on that PR I guess right now we can only use xcodebuild to generate docs with cross-product links using the experimental parameter? And we'd need to specify each dependency for each target so there's no practical way to enable this within Xcode at the moment?

**Comment by daltonclaybrook** [comment]
Tomorrow marks one year since the last update on this. Has there been any new progress you can speak to @d-ronnqvist?

**Comment by d-ronnqvist** [comment]
Yes. I made [this post in the Swift Forums][1] a few weeks ago with an announcement that you can try out an early, and still experimental, version of this.

I'm not quite sure when it would be appropriate to close this issue. With what's possible in the 6.0 release you can link to symbols in other modules if you pass an experimental feature flag. 

[1]: https://forums.swift.org/t/a-preview-of-doccs-support-for-combined-documentation/74373
